### PR TITLE
Adds Gradle task `dokkaHtml` to run Dokka document generation

### LINKED ISSUE #1457: Integrate Dokka into all SDK components and build API reference docs
**Describe the Issue**
Kotlin's documentation generation tool is called [Dokka](https://github.com/Kotlin/dokka). API reference docs should be available for developers, and rebuilding them should be a part of every release. Looking at the diff in generated reference docs between versions is a strong tool for understanding changes between versions. Code-level documentation is also tightly integrated with Android Studio, e.g. shows up when you hover over components in the code. Generating docs would also help identify where documentation is lacking or not keeping up with style guide expectations.

### PR DESCRIPTION
**Pull request by williamito**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1457

**Description**
Adds simple Gradle tasks to run Dokka document generation.

**Alternative(s) considered**
None; would love discussion on how to integrate doc generation better with current release processes however.

**Type**
Documentation

**Screenshots (if applicable)**
image

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Comment by jingtang10** [comment]
what's the command to be run to generate the docs?

**Comment by williamito** [comment]
You can generate docs like this for datacapture, you can specify the other subprojects as well:

./gradlew datacapture:dokkaHtml

Make sure to gradle sync successfully first, it seems a bit fickle about downloading Dokka from the plugins line sometimes.

File: datacapture/build.gradle.kts [hunk 1]
```diff
@@ -87,3 +91,21 @@ dependencies {
...
   androidTestImplementation(Dependencies.Espresso.espressoCore)
 }
+
+tasks.dokkaHtml.configure {
+  outputDirectory.set(file("../docs/datacapture"))
+  suppressInheritedMembers.set(true)
+  dokkaSourceSets {
+    named("main") {
+      moduleName.set("datacapture")
+      moduleVersion.set("0.1.0-beta03")
```
Comment (jingtang10) [inline datacapture/build.gradle.kts:101]: can this be linked from the `Releases.kt` file?

File: datacapture/build.gradle.kts [hunk 2]
```diff
@@ -87,3 +91,21 @@ dependencies {
   testImplementation(Dependencies.truth)
   androidTestImplementation(Dependencies.Espresso.espressoCore)
 }
+
+tasks.dokkaHtml.configure {
+  outputDirectory.set(file("../docs/datacapture"))
+  suppressInheritedMembers.set(true)
+  dokkaSourceSets {
+    named("main") {
+      moduleName.set("datacapture")
```
Comment (jingtang10) [inline datacapture/build.gradle.kts:100]: can this be linked from the `Releases.kt` file?

File: datacapture/build.gradle.kts [hunk 3]
```diff
@@ -1,8 +1,12 @@
+import java.net.URL
+
 plugins {
   id(Plugins.BuildPlugins.androidLib)
   id(Plugins.BuildPlugins.kotlinAndroid)
   id(Plugins.BuildPlugins.mavenPublish)
   jacoco
+  // Use Dokka 1.6.10 until https://github.com/Kotlin/dokka/issues/2452 is resolved.
+  id("org.jetbrains.dokka").version("1.6.10")
```
Comment (jingtang10) [inline datacapture/build.gradle.kts:9]: this is repeated in a few sub modules - can you put this in `Plugins.kt` so it's centralized?

File: engine/build.gradle.kts [hunk 4]
```diff
@@ -135,3 +138,21 @@ dependencies {
...
   testImplementation(Dependencies.truth)
 }
+
+tasks.dokkaHtml.configure {
+  outputDirectory.set(file("../docs/engine"))
+  suppressInheritedMembers.set(true)
+  dokkaSourceSets {
+    named("main") {
+      moduleName.set("engine")
+      moduleVersion.set("0.1.0-beta01")
```
Comment (jingtang10) [inline engine/build.gradle.kts:148]: ditto

File: workflow/build.gradle.kts [hunk 5]
```diff
@@ -139,3 +143,21 @@ dependencies {
...
   testImplementation(project(":testing"))
 }
+
+tasks.dokkaHtml.configure {
+  outputDirectory.set(file("../docs/workflow"))
+  suppressInheritedMembers.set(true)
+  dokkaSourceSets {
+    named("main") {
+      moduleName.set("workflow")
+      moduleVersion.set("0.1.0-alpha01")
```
Comment (jingtang10) [inline workflow/build.gradle.kts:153]: ditto

**Review by jingtang10 — CHANGES_REQUESTED** [review]
can the config block be made into a `Dokka.kt` file in the `buildSrc` directory? Examples:

https://github.com/google/android-fhir/blob/master/buildSrc/src/main/kotlin/JacocoConfig.kt
https://github.com/google/android-fhir/blob/master/buildSrc/src/main/kotlin/SpotlessConfig.kt
https://github.com/google/android-fhir/blob/master/buildSrc/src/main/kotlin/LicenseeConfig.kt

**Comment by williamito** [comment]
> can the config block be made into a `Dokka.kt` file in the `buildSrc` directory? Examples:
> 
> https://github.com/google/android-fhir/blob/master/buildSrc/src/main/kotlin/JacocoConfig.kt https://github.com/google/android-fhir/blob/master/buildSrc/src/main/kotlin/SpotlessConfig.kt https://github.com/google/android-fhir/blob/master/buildSrc/src/main/kotlin/LicenseeConfig.kt

I am not sure how to manage this, given how the Dokka plugin auto-creates tasks for every module it is added to. Would it be okay to merge having fixed your other comments?

**Review by jingtang10 — APPROVED** [review]
thanks william - the introduction of the util function in buildSrc will be done in my pr https://github.com/google/android-fhir/pull/1503

**Pull request merged**